### ISSUE TITLE
--prod modifier breaks dependsOn targets

### ISSUE THREAD

**Issue by Lonli-Lokli** [nrwl/nx#8562]
## Current Behavior
I have special target for running before build/serve, it works fine for non-prod builds but fails with --prod in Nx workspace.

## Expected Behavior
Build modifiers should not break 

## Steps to Reproduce
```
{
  "root": "apps/recon",
  "sourceRoot": "apps/recon/src",
  "projectType": "application",
  "targets": {
    "git-info": {
      "executor": "@nrwl/workspace:run-commands",
      "options": {
        "commands": [
          {
            "command": "npm run write-git-info",
            "forwardAllArgs": false
          }
        ]
      }
    },
    "build": {
      "executor": "@nrwl/web:webpack",
      "outputs": ["{options.outputPath}"],
      "options": {... },
      "dependsOn": [
        {
          "target": "git-info",
          "projects": "self"
        }
      ],
      "configurations": {
        "production": {...  }
      }
    },
    "serve": {...},
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["apps/recon/**/*.{ts,tsx,js,jsx}"]
      },
      "outputs": ["{options.outputFile}"]
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["coverage/apps/recon"],
      "options": {
        "jestConfig": "apps/recon/jest.config.js",
        "passWithNoTests": true
      }
    }
  },
  "tags": []
}

```

This command works
`npx nx build recon`

this command does not work
`npx nx build recon --prod`

### Failure Logs
**>  NX   ERROR  Cannot find configuration 'production' for project 'recon:git-info'**

### Environment

```
NX  Report complete - copy this into the issue template

  Node : 14.18.1
  OS   : win32 x64
  npm  : 6.14.15

nx : undefined
  @nrwl/angular : undefined
  @nrwl/cli : 13.4.3
  @nrwl/cypress : 13.4.3
  @nrwl/devkit : 13.4.3
  @nrwl/eslint-plugin-nx : 13.4.3
  @nrwl/express : 13.4.3
  @nrwl/jest : 13.4.3
  @nrwl/linter : 13.4.3
  @nrwl/nest : undefined
  @nrwl/next : undefined
  @nrwl/node : 13.4.3
  @nrwl/nx-cloud : undefined
  @nrwl/react : 13.4.3
  @nrwl/react-native : undefined
  @nrwl/schematics : undefined
  @nrwl/tao : 13.4.3
  @nrwl/web : 13.4.3
  @nrwl/workspace : 13.4.3
  @nrwl/storybook : 13.4.3
  @nrwl/gatsby : undefined
  typescript : 4.5.4
  rxjs : 6.6.7
```

**Comment by KosnIre** [comment]
> Cannot find configuration 'production' for project 'recon:git-info'

What happens if you add: 

```json
"configurations": {
    "production": {}
}
```

to the `"git-info"` target?

**Comment by Lonli-Lokli** [comment]
It will work but it feels redundunt to have all this in pure target

**Comment by linbudu599** [comment]
> It will work but it feels redundunt to have all this in pure target

I think this behavior is expected because when you build a project using the production configuration, the dependencies of that project should also be built using the production configuration as a corresponding way. Also, adding a new determination to the nx source code whether a project is built as a dependency and correspondingly allowing the production configuration to be ignored may add unnecessary complexity

**Comment by Lonli-Lokli** [comment]
@linbudu599 what are other options for targets except dependencies which will not require same configuration?

**Comment by KosnIre** [comment]
It does feel a little verbose to have to specify an empty config for a target that doesn't need it, but it could also prevent human-error. It's like nx is saying "You told me to use the production config, but I don't see a production config".

@Lonli-Lokli  It sounds to me like you're saying you want to specify your production config just once and have nx forward it to all targets that the initial target depends on. If all of your targets require the same production configuration settings, then you may want to consider moving those settings to environment variables. That way each script could read from them.

However, I'm not sure if that works for the `"executor": "@nrwl/web:webpack",`, so you would have to specify the config at most twice. As a feature request, it might be handy to be able to do something like:

```json
"configurations": {
  "production": {
    "optimization": "{env.optimization}",
    "sourceMap": "{env.sourceMap}",
  }
}
```

in project.json

**Comment by vsavkin** [comment]
Thank you for submitting the issue!

It has been fixed here: https://github.com/nrwl/nx/commit/ccda8cf8160a50edcda99944d589999916eba337 and will be released as part of 13.5.0 tomorrow.

Thank you!
